BUILD

```diff
@@ -0,0 +1,247 @@
+        "@com_github_aws_aws_sdk_go_v2_feature_s3_manager//:manager",
+        "@com_github_aws_aws_sdk_go_v2_service_s3//:s3",
+        "@com_github_gobuffalo_packr_v2//:packr",
+        "@com_github_gorilla_mux//:mux",
+        "@com_github_pkg_errors//:errors",
+        "@com_github_prometheus_client_golang//prometheus",
+        "@com_github_prometheus_client_golang//prometheus/promhttp",
+        "@com_github_satori_go_uuid//:go_uuid",
+        "@com_github_sirupsen_logrus//:logrus",
+        "@com_github_spf13_pflag//:pflag",
+        "@com_github_spf13_viper//:viper",
+    ],
+)
+
+#
+# Tests
+#
+
+go_test(
+    name = "ansible_test",
+    srcs = ["ansible_test.go"],
+    data = [
+        ":ansible-puller.json",
+        ":testdata",
+    ],
+    embed = [":ansible_puller_lib"],
+    deps = [
+        "@com_github_stretchr_testify//assert",
+        "@com_github_stretchr_testify//suite",
+    ],
+)
+
+go_test(
+    name = "ansible_puller_test",
+    srcs = [
+        "ansible_test.go",
+        "http_downloader_test.go",
+        "http_test.go",
+        "s3_downloader_test.go",
+        "unarchive_test.go",
+    ],
+    data = [
+        ":ansible-puller.json",
+        ":testdata",
+    ],
+    embed = [":ansible_puller_lib"],
+    deps = [
+        "@com_github_stretchr_testify//assert",
+        "@com_github_stretchr_testify//suite",
+    ],
+)
+
+go_test(
+    name = "http_downloader_test",
+    srcs = ["http_downloader_test.go"],
+    data = [
+        ":ansible-puller.json",
+        ":testdata",
+    ],
+    embed = [":ansible_puller_lib"],
+    deps = [
+        "@com_github_stretchr_testify//assert",
+        "@com_github_stretchr_testify//suite",
+    ],
+)
+
+go_test(
+    name = "http_test",
+    srcs = ["http_test.go"],
+    data = [
+        ":ansible-puller.json",
+        ":testdata",
+    ],
+    embed = [":ansible_puller_lib"],
+    deps = [
+        "@com_github_stretchr_testify//assert",
+        "@com_github_stretchr_testify//suite",
+    ],
+)
+
+go_test(
+    name = "s3_downloader_test",
+    srcs = ["s3_downloader_test.go"],
+    data = [
+        ":ansible-puller.json",
+        ":testdata",
+    ],
+    embed = [":ansible_puller_lib"],
+    deps = [
+        "@com_github_stretchr_testify//assert",
+        "@com_github_stretchr_testify//suite",
+    ],
+)
+
+go_test(
+    name = "unarchive_test",
+    srcs = ["unarchive_test.go"],
+    data = [
+        ":ansible-puller.json",
+        ":testdata",
+    ],
+    embed = [":ansible_puller_lib"],
+    deps = [
+        "@com_github_stretchr_testify//assert",
+        "@com_github_stretchr_testify//suite",
+    ],
+)
+
+#
+# deb
+#
+
+go_binary(
+    name = "ansible_puller_bin",
+    basename = "ansible-puller",
+    embed = [":ansible_puller_lib"],
+    visibility = ["//visibility:public"],
+)
+
+pkg_tar(
+    name = "pkg_tar_ansible_puller_bin",
+    srcs = [":ansible_puller_bin"],
+    mode = "0755",
+    package_dir = "/usr/bin",
+)
+
+pkg_tar(
```

Comment:
Should we be including this file? I don't think the config should be updated by default here. Maybe make this file `/etc/ansible-puller/ansible-puller.example.json`?

Reply:
good idea